tests/unit/services/auth/test_auth_handlers.py

```
@@ -0,0 +1,329 @@
+---------------------------------
+Tests for auth_handlers.py - OAuth2 callback processing.
+"""
+
+import io
+from threading import Event, Lock, RLock
+from unittest.mock import Mock, patch, MagicMock, PropertyMock
+
+import pytest
+
+
+def create_mock_handler(path: str, callback_event: Event = None, callback_lock: Lock = None, callback_data: dict = None):
```

Ruff (0.14.10)

PEP 484 prohibits implicit `Optional`

Convert to `Optional[T]`

(RUF013)